core/eval.go

```diff
@@ -173,6 +173,24 @@ func evalJSONSET(args []string) []byte {
 	// Note: args[1] (path) is ignored in this implementation
 	jsonStr := args[2]
 
+	for i := 3; i < len(args); i++ {
+		switch args[i] {
+		case "NX", "nx":
```

Comment:
We should also handle the case:
`JSON.SET user $ '{"name":"John","age":30}' NX XX`

Below is the output from REDIS:
```
redis:6379> JSON.SET user $ '{"name":"John","age":30}' NX XX
(error) ERR unknown command 'JSON.SET', with args beginning with: 'user' '$' '{"name":"John","age":30}' 'NX' 'XX'
```

Reply:
Yes, will raise a separate pr for this.

Reply:
Although, currently even dice db SET , EXPIRE command does not have handling for this case.
@arpitbbhayani @JyotinderSingh What's your suggestion, should we handle this case ?

If yes, we can do something like this
```
key := args[0]
path := args[1]
jsonStr := args[2]

pattern := `^(?i)(NX|XX)$`
re, err := regexp.Compile(pattern)
if err != nil {
     return RESP_NIL
}

if len(args) > 3 && !re.MatchString(strings.Join(args[3:], " ")) {
     return Encode(errors.New("ERR syntax error"), false)
}
```

We can also add a regex validation for each command before executing it.

Reply:
Let's handle this, but try to not use regex since it might not be efficient to run a regex match on every request

Reply:
We are handling it for EX/PX options for the set command. Please check [https://github.com/DiceDB/dice/blob/c00484a47598685eba70333cd15fecd36a5924be/core/eval.go#L125](here)

Reply:
@JyotinderSingh @soumya-codes please review.
https://github.com/DiceDB/dice/pull/260